syft/frameworks/torch/he/fv/key_generator.py

```diff
@@ -30,17 +30,17 @@ def keygen(self):
         """
         self._generate_sk()
         self._generate_pk()
-        return [self._secret_key, self._public_key]
+        return [self.secret_key, self.public_key]
 
-    def _generate_sk(self, is_initialized=False):
-        param = self._context.param
-
-        if not is_initialized:
-            self._secret_key = SecretKey(sample_poly_ternary(param))
```

Comment:
Why no more init checking?

Reply:
`is_initialized` is no more needed because if the context is not set up it will throw the error while validation in context class. So the secret key can be created using context(anytime with validated context object). And for other keys, I will check for the None value of secret key.